netbox/dcim/tests/test_views.py

```diff
@@ -1907,6 +1954,49 @@ def test_module_component_adoption(self):
         # Check that the Interface now has a module
         self.assertIsNotNone(interface.module)
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=['*'])
+    def test_module_bulk_adoption(self):
+        self.add_permissions('dcim.add_module')
+
+        interface_name = "Interface-1"
+
+        # Add an interface to the ModuleType
+        module_type = ModuleType.objects.first()
+        InterfaceTemplate(module_type=module_type, name=interface_name).save()
+
+        form_data = self.form_data.copy()
+        device = Device.objects.get(pk=form_data['device'])
+
+        # Create an interface to be adopted
+        interface = Interface(device=device, name=interface_name, type=InterfaceTypeChoices.TYPE_10GE_FIXED)
+        interface.save()
+
+        # Ensure that interface is created with no module
+        self.assertIsNone(interface.module)
+
+        # Create a module with adopted components
```

Comment:
One small thing, this testcase actually fails on my PC. It's trying to install in the first modulebay of the device, which is already occupied from the setup. If I change to another bay it passes. Not sure if it's something with my setup or from the rebase.

Reply:
Same. Maybe related to 9ef24d3f43022f61044756741f40e095e9290bc1? In any case, changing it to an empty bay appears to fix the test.

Reply:
Pushed a fix for this in [2a55c29](https://github.com/netbox-community/netbox/pull/9498/commits/2a55c29a9a09fe20a61dcc9964e298769ada3e72).